Blocks/Persistence/file.h

```diff
@@ -192,8 +199,16 @@ class FilePersister {
                 }
                 head = us;
                 head_offset = std::streamoff(current_offset) + offset;
-              } else {
-                head_offset = 0;
+              } else if (!value.compare(0, signature_key_length, constants::kSignatureDirective)) {
+                // The signature could be only at the beginning of the file.
```

Comment:
Nit: `s/could be only/could only be/`, or "should only be", or "The signature, if present, should be at ...".